### PR TITLE
London Class 7 - Aisha Khan - JavaScript Core 2 - Week 1

### PR DESCRIPTION
**Pull request by aishak05**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: mandatory/2-water-bottle.js [hunk 1]
```diff
@@ -23,18 +23,37 @@ You have to implement the missing features according to the specification.
 let bottle = {
   volume: 0,
   fillUp: function () {
+    return this.volume = this.volume + 100-this.volume;
     // calling this function should completely fill your bottle (volume = 100);
   },
   pour: function () {
+    if(this.volume < 90){
```
Comment (kmh256) [inline mandatory/2-water-bottle.js:30]: Great solution! 
You could also do a shorthand version:
```
if (this.volume <= 90) {
   return this.volume += 10;
}
```

File: mandatory/2-water-bottle.js [hunk 2]
```diff
@@ -23,18 +23,37 @@ You have to implement the missing features according to the specification.
...
+    return this.volume = this.volume + 10;
+    } else {
+      return this.volume
+    }
+    
     // calling this function should increase your bottle volume by 10 units;
   },
   drink: function () {
+    if(this.volume >= 10){
```
Comment (kmh256) [inline mandatory/2-water-bottle.js:39]: Great solution! 
You could also do a shorthand version:
```
if (this.volume >= 90) {
   return this.volume -= 10;
}
```

**Review by kmh256 — COMMENTED** [review]
Well done 👍 great solutions,  just a couple of suggestions.

File: mandatory/1-writers.js [hunk 3]
```diff
@@ -69,10 +74,22 @@ Exercise 2:
   "Writer {firstName} {lastName} died at {age} years old."
 */
 
+writers.filter(element => {
+  if(element.age > 40 && element.age <49 && element.alive === false){
+    console.log(`Writer ${element.firstName} ${element.lastName} died at ${element.age} years old.`);
+  }
+})
```
Comment (syedur-r) [inline mandatory/1-writers.js:81]: Great work on adding multiple conditions to filter out the age of writers.

Although the outputs are correct, you may want to have another look at the instructions which says to check age range between 40 and 49 (including 40 and including 49). Your age condition excludes 40 year old writers and 49 year old writers, and only checks 41 to 48 year old writers. How would you include 40 and 49 in your condition?

File: mandatory/1-writers.js [hunk 4]
```diff
@@ -69,10 +74,22 @@ Exercise 2:
...
   Only `console.log()` out alive writers who are in their 40s (meaning between 40 and 49):
 
   "Hi, my name is {firstName} {lastName}. I am {age} years old."
 */
+
+writers.filter(element => {
+  if(element.age > 40 && element.age <49 && element.alive === true){
+    console.log(`Hi, my name is ${element.firstName} ${element.lastName}. I am ${element.age} years old.`);
+  }
+})
```
Comment (syedur-r) [inline mandatory/1-writers.js:95]: The same applies here, how would you include the age 40 and 49 in your condition?

File: mandatory/4-people-I-know.js [hunk 5]
```diff
@@ -425,6 +449,13 @@ This time, I only want the full names of the people who can multitask
 
 let friendsWhoCanMultitask = [];
```
Comment (11ma) [inline mandatory/4-people-I-know.js:451]: Beautifully done Aisha, well done this works and great use of the `for...of` loop. Nicely done 👏

**Pull request closed without merging**